### ISSUE TITLE
REL: list contributors using GitHub handles

### ISSUE THREAD

**Issue by tupui** [scipy/scipy#15543]
When we do the release on GitHub, I guess we copy/paste the text we put in the rst files. If yes, then can we adjust the contributor list to use the proper GitHub handles? Because right now GitHub only list/link a few contributors instead of the >100 contributors who effectively contributed. It would help the recognition of smaller contributors.

As we can see in the screenshot, only if you use the GH handle, the UI will pick it up.

<img width="426" alt="Screenshot 2022-02-07 at 13 22 20" src="https://user-images.githubusercontent.com/23188539/152787310-528aca70-e460-4a83-a279-75cf40bf838d.png">

**Comment by tylerjereddy** [comment]
> GitHub only list/link a few contributors instead of the >100 contributors who effectively contributed.

We list all of the contributors using the release scripts and `.mailmap` machinery. They are clearly visible on the release page:

![image](https://user-images.githubusercontent.com/7903078/152817550-2b4451d2-516f-4004-b8b1-7c03b5fda117.png)

etc..

The separate GitHub feature that automatically parses out contributors to produce a separate list could be taken advantage of, though I'd actually prefer to go in the other direction--not use GitHub handles at all and stick to the "real names" for folks who share them, at least in the short term it is fair bit less effort--we only have 1 or 2 handles in there and they are mostly accidental.

**Comment by rgommers** [comment]
I agree with @tylerjereddy here. The list of names is generated with:
```
python tools/authors.py v1.7.0..v1.8.0 
```
The aim is to include people's real names, and for consistency we shouldn't change that. For people who do not want their real name included, we use GitHub handles. Here all that is needed is adjust `.mailmap` to ensure the `@` is removed from `@endolith`'s name.

**Comment by tupui** [comment]
Ok, If we stick with the current solution, I can change the script to remove `@`.

@tylerjereddy we discussed a bit the topic during today's meeting. I will summarise my whole intent/idea here.

I see two points of improvements on the release to give contributors (especially small ones) credit.

1. Make sure that everyone in the list has it's GitHub handle. I would add it instead of removing it, since GitHub seems to care about it and create a special section for that. Another possibility is to not put the whole list as other projects are doing (pandas, sklearn for instance. ex. https://github.com/scikit-learn/scikit-learn/releases/tag/1.0.2). They just give a 2 line install instruction and redirect to the Changelog. It's actually easier to navigate and see the dates of releases.
2. Improve the Changelog itself. Looking at sklearn example again, to me, what they have is great in terms of completeness of the information, clarity, presentation and also attribution. All the points are listed with direct links to PR and the authors name is right there which clearly gives a sense of the amount/type of work done by different people. I don't know if sklearn is doing this work manually though and I agree that we don't want to have more work to do with our limited resources 😃 We do have this information right now but scattered. Except for the "quantity/type" which is not visible without going through the PRs/issues full list on the long version of the changelog, this could also be an "easy" fix to list the PRs/issues and the author right next to them (happy to have a look too if we want to do this).

**Comment by rgommers** [comment]
@tupui points (1) and (2) are very different. You started with only (1), which (as you mentioned in the meeting) is a _very minor thing_. Please let's just take the lowest-effort solution here, remove the single `@`, and be done with it.

(2) is different. The scikit-learn Changelog is quite nice, but it's unclear how it is generated. I imagine it's a significant PR workflow change, perhaps with a bot, or giving extra work for the release manager. If you're interested in upgrading to such a Changelog, it seems like you should investigate how that is generated. If it is not significantly more work we may want to do that, but it's separate from your original bug report.

**Comment by tupui** [comment]
I will remove `@`.

I talked about the point 2 to be transparent about the discussion we had. I will investigate to see if there is an easy path.
